### ISSUE TITLE
Compact Persisted Chunks Lifecycle Policy

### ISSUE THREAD

**Issue by tustvold** [influxdata/influxdb_iox#2668]
Currently the LifecyclePolicy will never compact persisted chunks, however, in the case of a chunk with a hard delete predicate it should request compaction to perform this delete.

Additional critieria that might trigger a persisted compaction:

* Overlapping timestamp ranges
* Too many soft delete predicates

This ticket tracks devising and implementing such a policy.

Depends on #2665

**Comment by NGA-TRAN** [comment]
Close this because it is duplicate of #3355
